packages/discord.js/src/structures/SelectMenuComponent.js

```diff
@@ -1,11 +1,64 @@
 'use strict';
 
-const { SelectMenuComponent: BuildersSelectMenuComponent } = require('@discordjs/builders');
-const Transformers = require('../util/Transformers');
+const Component = require('./Component');
 
-class SelectMenuComponent extends BuildersSelectMenuComponent {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+/**
+ * Represents a select menu component
+ * @extends {Component}
+ */
+class SelectMenuComponent extends Component {
+  /**
+   * The placeholder for this select menu
+   * @type {?string}
+   * @readonly
+   */
+  get placeholder() {
+    return this.data.placeholder ?? null;
+  }
+
+  /**
+   * The maximum amount of options that can be selected
+   * @type {?number}
+   * @readonly
+   */
+  get maxValues() {
+    return this.data.max_values ?? null;
+  }
+
+  /**
+   * The minimum amount of options that must be selected
+   * @type {?number}
+   * @readonly
+   */
+  get minValues() {
+    return this.data.min_values ?? null;
+  }
+
+  /**
+   * The custom id of this select menu
+   * @type {string}
+   * @readonly
+   */
+  get customId() {
+    return this.data.custom_id;
+  }
+
+  /**
+   * Whether this select menu is disabled
+   * @type {?boolean}
+   * @readonly
+   */
+  get disabled() {
+    return this.data.disabled ?? null;
+  }
+
+  /**
+   * The options in this select menu
+   * @type {APISelectMenuOption[]}
```

Comment:
This doesn't seem ideal for a d.js component, shouldn't it return camelCase data

Reply:
`APISelectMenuOption` has camelCased properties already.